be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -215,6 +217,30 @@ StatusOr<DriverState> PipelineDriver::process(RuntimeState* runtime_state) {
     }
 }
 
+void PipelineDriver::maybe_short_circuit() {
+    size_t new_first_unfinished = _first_unfinished;
+    for (int i = _first_unfinished; i < _operators.size() - 1; i++) {
+        if (_operators[i]->is_finished()) {
+            new_first_unfinished = i + 1;
```

Comment:
need break?

Reply:
There may be multiple HashJoinProbeOperators in one driver, so we need find the last short-circuitedable operator.